engine/src/main/scala/cromwell/engine/workflow/tokens/TokenPool.scala

```diff
@@ -30,11 +30,13 @@ object TokenPool {
 
 final case class SizedTokenPool(pool: Set[JobExecutionToken], override val currentLoans: Set[JobExecutionToken]) extends TokenPool {
 
-  override def pop(): TokenPoolPop = pool.toList match {
-    case head :: tail => TokenPoolPop(SizedTokenPool(tail.toSet, currentLoans + head), Some(head))
-    case Nil => TokenPoolPop(SizedTokenPool(Set.empty, currentLoans), None)
+  override def pop(): TokenPoolPop = if (pool.isEmpty) {
+    TokenPoolPop(SizedTokenPool(Set.empty, currentLoans), None)
+  } else {
+    TokenPoolPop(SizedTokenPool(pool.tail, currentLoans + pool.head), pool.headOption)
```

Comment:
I did some refactoring but I really don't like having to iterate the entire set just to remove the head. Since nothing else looks at it, I'm just going to make it a list so I can use `::` to split it!